lib/portability.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+  Copyright (C) 2024  Abe Tomoaki <abe@clear-code.com>
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <algorithm>
+#include "grn.h"
+#include "grn_portability.h"
```

Comment:
最初にGroonga関連のヘッダーファイルを読んでから標準のヘッダーファイルを読もうか。たまに`grn.h`とかで`#define`して挙動を変えているんだよ。

```suggestion
#include "grn.h"
#include "grn_portability.h"

#include <algorithm>
```